mono/mini/mixed_callstack_plugin.c

```diff
@@ -1,29 +1,71 @@
 #include "mixed_callstack_plugin.h"
 #include "mono/metadata/mono-debug.h"
 #include "mono/metadata/profiler.h"
+#include "mono/mini/seq-points.h"
 
 #if !defined(DISABLE_JIT) && defined(HOST_WIN32)
 
 static gboolean enabled;
 static mono_mutex_t mutex;
-static HANDLE fileHandle;
-int pmipFileNum;
+static GPtrArray* domainHandles;
+static gboolean legacyMode;
 
 #define mixed_callstack_plugin_lock() mono_os_mutex_lock (&mutex)
 #define mixed_callstack_plugin_unlock() mono_os_mutex_unlock (&mutex)
 
 static void mixed_callstack_plugin_on_domain_unload_end (MonoProfiler *prof, MonoDomain *domain);
 
-void
-create_next_pmip_file()
+typedef struct {
+	HANDLE fileHandle;
+	gint32 domain_id;
+	gint32 count;
+} DomainHandle;
+
+DomainHandle*
+get_domain_handle(gint32 domain_id)
 {
-	char* file_name = g_strdup_printf("pmip_%d_%d.txt", GetCurrentProcessId(), pmipFileNum++);
-	char* path = g_build_filename(g_get_tmp_dir(), file_name, NULL);
-	char* version = "UnityMixedCallstacks:1.0\n";
+	DomainHandle* dHandle = NULL;
+	domain_id = legacyMode ? mono_get_root_domain()->domain_id : domain_id;
+	for (int i = 0; i < domainHandles->len; i++)
+	{
+		if (((DomainHandle*)domainHandles->pdata[i])->domain_id == domain_id)
+		{
+			dHandle = (DomainHandle*)domainHandles->pdata[i];
+			break;
+		}
+	}
+
+	return dHandle;
+}
+
+DomainHandle*
+create_next_pmip_file(gint32 domain_id)
+{
+	char* file_name;
+	char* path;
+	char* version = "UnityMixedCallstacks:2.0\n";
 	long bytesWritten = 0;
+	int pmipFileNum = 1;
 
 	mixed_callstack_plugin_lock ();
 
+	HANDLE fileHandle = NULL;
+	DomainHandle* dHandle = get_domain_handle(domain_id);
+
+	if (dHandle)
+	{
+		fileHandle = dHandle->fileHandle;
+		pmipFileNum = ++dHandle->count;
```

Comment:
Why do we need this? Because domain_id's get reused?

Reply:
Yes I believe that was the original reason why we have a count value in the filename. It was there previously and I kept it. It makes it a little easier to be sure when debugging on the other side that you're dealing with a new domain.